google/resource_container_cluster.go

```diff
@@ -698,6 +699,31 @@ func resourceNodeConfigEmptyGuestAccelerator(diff *schema.ResourceDiff, meta int
 	return nil
 }
 
+func resourceContainerClusterNodeVersionCustomizeDiff(diff *schema.ResourceDiff, meta interface{}) error {
+	// separate func to allow unit testing
+	return resourceContainerClusterNodeVersionCustomizeDiffFunc(diff)
+}
+
+// resourceContainerClusterNodeVersionCustomizeDiffFunc is checking that argument node_version have
+// same value as min_master_version if setted on create.
+func resourceContainerClusterNodeVersionCustomizeDiffFunc(diff TerraformResourceDiff) error {
+
+	// By pass the diff if we are not on create.
+	oldValueName, _ := diff.GetChange("name")
+	if oldValueName != "" {
+		return nil
+	}
+
+	_, newValueNode := diff.GetChange("node_version")
```

Comment:
Can you compare these values by splitting off the GKE part for consistency with the apply-time check? https://github.com/kdefives/terraform-provider-google/blob/feature-i2701/google/resource_container_cluster.go#L807-L808

While this is a good change (it's a better validation) it's technically a breaking one since the validations are not 1:1.